js/controller.js

```diff
@@ -204,12 +252,15 @@
             //Track when the Annyang is listening to us
             AnnyangService.start(function(listening){
                 $scope.listening = listening;
+                $scope.startAutoSleepTimer();
             }, function(interimResult){
                 $scope.interimResult = interimResult;
                 $timeout.cancel(resetCommandTimeout);
+                $scope.startAutoSleepTimer();
             }, function(result){
                 $scope.interimResult = result[0];
                 resetCommandTimeout = $timeout(restCommand, 5000);
+                $scope.startAutoSleepTimer();
             });
```

Comment:
You really should only be calling `$scope.startAutoSleepTimer()` in the result callback. Doing it for listening will cause it to fire all the time (therefore defeating the purpose of the timer in the first place). You also shouldn't be calling it in the interimResult callback because you'll just be spamming reset.

This does leave the *very* slim chance that the timer will expire before a final result is recognized and the mirror would go to sleep when you are speaking, but I highly doubt this will ever actually happen.